ciborium/src/value/de.rs

```diff
@@ -124,7 +124,9 @@ impl<'de> serde::de::Visitor<'de> for Visitor {
 
     #[inline]
     fn visit_map<A: de::MapAccess<'de>>(self, mut acc: A) -> Result<Self::Value, A::Error> {
-        let mut map = Vec::<(Value, Value)>::with_capacity(acc.size_hint().unwrap_or(0));
+        let mut map = Vec::<(Value, Value)>::with_capacity(
+            acc.size_hint().filter(|&l| l < 1024).unwrap_or(0),
```

Comment:
Description says:
> Only preallocate up to 1024 Value objects.

But this code will not allocate at all when size hint >= 1024.